### PR TITLE
Webdriver tests should default to use the HTTPS protocol

### LINKED ISSUE #28847: Webdriver tests should default to use the HTTPS protocol
As just noticed when trying to get a test with `Cross-Origin-Opener-Policy` added, the [`url` fixture](https://github.com/web-platform-tests/wpt/blob/bae534da5146c3504805c49d0c55b61c8f96ffbd/webdriver/tests/support/fixtures.py#L226) (which other fixtures like [`inline`](https://github.com/web-platform-tests/wpt/blob/bae534da5146c3504805c49d0c55b61c8f96ffbd/webdriver/tests/support/fixtures.py#L321) are based on) defaults to HTTP. That means we basically don't test anything for HTTPS right now, which is IMO a huge deficit. 

Switching the default to `HTTPS` raises another question. How can we get a valid certificate added to the certificate store of each browser to not have to enable the `acceptInsecureCerts` capability? As @jgraham pointed out for Firefox it will be easy soon when his patch series from PR #28729. 

But how does it look like for other browsers?

CC'ing @foolip, @burg, @JohnChen0, @bwalderman.

### PR DESCRIPTION
**Pull request by whimboo**
This converts all the usages of HTTP to HTTPS by default. Some tests will still use HTTP to test non-secure connections like the navigation tests.

Closes #28847.

### CODE REVIEW

File: webdriver/tests/bidi/browsing_context/navigate/invalid.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ async def test_params_url_invalid_type(bidi_session, new_tab, value):
         )
 
 
-@pytest.mark.parametrize("value", ["http://:invalid", "http://#invalid"])
+@pytest.mark.parametrize("value", ["https://:invalid", "https://#invalid"])
```
Comment (jrandolf) [inline webdriver/tests/bidi/browsing_context/navigate/invalid.py:31]: This doesn't really matter here. It'd be nice to add `http` as well here.

**Review by jrandolf — APPROVED** [review]

File: webdriver/tests/bidi/browsing_context/navigate/invalid.py (hunk 1 above)
Comment (whimboo) [reply webdriver/tests/bidi/browsing_context/navigate/invalid.py:31]: Sure. Can add another parameterized value for protocol.

**Comment by whimboo** [comment]
@jgraham and @patrickangle mind reviewing? Thanks.

**Review by juliandescottes — APPROVED** [review]
FYI, ran this through mozilla's CI and it seems fine: https://treeherder.mozilla.org/jobs?repo=try&revision=4c9d23c67edb1c639182d96380b7bfd56400dcd2

**Comment by patrickangle** [comment]
Seems reasonable to me. @gsnedders any concerns from our end I'm not considering?

**Review by patrickangle — APPROVED** [review]

**Comment by gsnedders** [comment]
> Looking at the Safari results, I think they're consistent with flakiness rather than the change breaking anything.

Yeah, looking [at this with more runs](https://wpt.fyi/results/webdriver/tests/classic?q=none%28status%3Amissing%29%20is%3Adifferent&run_id=5147719797047296&run_id=6225851413168128&run_id=5082951824441344&run_id=5081179965882368&run_id=5077349173821440&run_id=5179373773127680&run_id=5137157398724608&run_id=5174413220118528&run_id=5184080000319488&run_id=5202784717111296&run_id=5080930102804480&run_id=5138419397689344&run_id=5114007122345984&run_id=5131751863615488&run_id=6229001620684800), it very much just looks like flakiness rather than any changes.

> Seems reasonable to me. @gsnedders any concerns from our end I'm not considering?

Maybe. I'll try and check later today.

**Review by gsnedders — APPROVED** [review]
OK, having actually checked, this is totally fine from our side.

**Comment by whimboo** [comment]
Thanks everyone! Given that there are test failures I cannot get it merged myself. @gsnedders or @jgraham could you please admin merge? Thanks!

**Pull request merged**